tlatools/src/tlc2/util/DotStateWriter.java

```diff
@@ -37,15 +47,60 @@
  */
 public class DotStateWriter extends StateWriter {
 
-	public DotStateWriter(final String fname) throws IOException {
-		this(fname, "strict ");
+	// The Graphviz color scheme that is used for state transition edge colors. See
+	// https://www.graphviz.org/doc/info/colors.html for more details on color schemes.
+	static private final String dotColorScheme = "paired12";
+	
+	// Maximum number of unique colors. Determined by the Graphviz color scheme that is used.
+	static private final Integer colorGenMax = 12;
+	
+	// If the total number of states in the resulting graph exceeds this count, we
+	// omit action labels no matter what (even if the option was specified as
+	// 'true'). The edge labels add considerable visual clutter for graphs of
+	// non-trivial sizes, and so we aim to not make graphs unreadable. The choice of
+	// this threshold is somewhat arbitrary, but it serves the right purpose i.e.
+	// include edge labels on only the smallest of state graphs.
+	static private final Integer maxNumStatesForActionLabels = 35;
```

Comment:
This is the parameter that I added but did not include logic for. Wanted to see if you had any suggestions on the simplest way to implement this.